### DISCUSSION TITLE
How many concurent user can the bot handle?

### DISCUSSION THREAD

**Discussion by nacereddinedjebar** [rayanfer32/TgindexPro#27] [Q&A] [answered]
Dear Devs,

if i use the direct link that the bot generated for some files for example and access stream it to vlc player and let's suppose that there is 1000 user are watching the same movie, will that affect the bot server behaviour or all the load are on the telegram server?

**Comment by pseudokawaii** [comment, marked as answer]
Of course that will affect your app server first, Your app is caching the contents on its server first, then serving them to the end users. So if a lot of users are requesting the same file at the same time, Your app instance has to be capable enough to handle all that load. Also, even if your server is capable enough, tg-index itself is not capable of parallel transfer yet, so your users will face buffering issues quite often.

As for Telegram, it uses CDN to deliver it's content, and it's designed so that a lot of users can request the same content at the same time. But when you use tg-index, you are actually serving all the content that's available to you from Telegram using a single API key, so if you make too many requests at the same time, it will most probably affect your account too as I'm pretty sure that Telegram throttles outbound transfer speed per account/session.
